File: AsptNet Core 2/TaskList/SharedKernel.Domain/Modelo/PageResult.cs
using System.Collections.Generic;
using SharedKernel.Domain.Modelo.Interface;

namespace SharedKernel.Domain.Modelo
{
    public class PageResult<T> where T : IEntity
    {
        public int Page { get; set; }
        public int PageSize { get; set; }
        public int TotalItems { get; set; }
        public int TotalPages { get; set; }

        public string PreviousPage { get; set; }
        public string NextPage { get; set; }

        public IList<T> Data { get; set; }
    }
}

File: Angular2/src/app/base/model/pageResult.ts
import { Observable } from 'rxjs/Observable';

export interface IPageResult {
    page: number;
    pageSize: number;
    totalItems: number;
    totalPages: number;
    previousPage: string;
    nextPage: string;
    data: Observable<any[]>;    
}

export class PageResult implements IPageResult{
    constructor(public page: number,
                public pageSize: number,
                public totalItems: number,
                public totalPages: number,
                public previousPage: string,
                public nextPage: string,
                public data: Observable<any[]>){
    }
}